### PR TITLE
Strict 2FA

### PR DESCRIPTION
**Pull request by rullzer**
As discussed @ChristophWurst 

Todo:
- [ ] Notify all 2FA apps
- [x] Add to https://github.com/nextcloud/server/issues/7827

### CODE REVIEW

File: lib/public/Authentication/TwoFactorAuth/IProvider.php [hunk 1]
```diff
@@ -65,7 +66,7 @@ public function getDisplayName();
 	 *
 	 * @return string
 	 */
-	public function getDescription();
+	public function getDescription(): string	;
```
Comment (ChristophWurst) [inline lib/public/Authentication/TwoFactorAuth/IProvider.php]: trailing whitespace :wink:

**Review by ChristophWurst — CHANGES_REQUESTED** [review]
Awesome :heart_eyes:

File: lib/public/Authentication/TwoFactorAuth/IProvider.php (hunk 1 above)
Comment (rullzer) [reply lib/public/Authentication/TwoFactorAuth/IProvider.php]: where...

**Review by ChristophWurst — APPROVED** [review]
:rocket:

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**